statsmodels/discrete/discrete_model.py

```diff
@@ -1173,6 +1177,374 @@ def hessian(self, params):
+        mu_p = np.power(mu, p)
+        a1 = 1 + alpha * mu_p
+        a2 = mu + alpha * mu_p * y
+        a3 = alpha * p * mu ** (p - 1)
+        a4 = a3 * y
+        a5 = p * mu ** (p - 1)
+        dmudb = mu * exog
+
+        # for dl/dparams dparams
+        dim = exog.shape[1]
+        hess_arr = np.empty((dim+1,dim+1))
+
+        for i in range(dim):
+            for j in range(i + 1):
+                hess_arr[i,j] = np.sum(mu * exog[:,i,None] * exog[:,j,None] *
+                    (mu * (a3 * a4 / a1**2 - 2 * a3**2 * a2 / a1**3 + 2 * a3 *
+                    (a4 + 1) / a1**2 - a4 * p / (mu * a1) + a3 * p * a2 /
+                    (mu * a1**2) + a4 / (mu * a1) - a3 * a2 / (mu * a1**2) +
+                    (y - 1) * a4 * (p - 1) / (a2 * mu) - (y - 1) *
+                    (1 + a4)**2 / a2**2 - a4 * (p - 1) / (a1 * mu) - 1 /
+                    mu**2) + (-a4 / a1 + a3 * a2 / a1**2 + (y - 1) *
+                    (1 + a4) / a2 - (1 + a4) / a1 + 1 / mu)), axis=0)
+        tri_idx = np.triu_indices(dim, k=1)
+        hess_arr[tri_idx] = hess_arr.T[tri_idx]
+
+        # for dl/dparams dalpha
+        dldpda = np.sum((2 * a4 * mu_p / a1**2 - 2 * a3 * mu_p * a2 / a1**3 -
+                        mu_p * y * (y - 1) * (1 + a4) / a2**2 + mu_p *
+                        (1 + a4) / a1**2 + a5 * y * (y - 1) / a2 - 2 *
+                        a5 * y / a1 + a5 * a2 / a1**2) * dmudb,
+                        axis=0)
+
+        hess_arr[-1,:-1] = dldpda
+        hess_arr[:-1,-1] = dldpda
+
+        # for dl/dalpha dalpha
+        dldada = mu_p**2 * (3 * y / a1**2 - y**2 * (y - 1) / a2**2 - 2 * a2 /
+                            a1**3)
+
+        hess_arr[-1,-1] = dldada.sum()
+
+        return hess_arr
+
+    def predict(self, params, exog=None, exposure=None, offset=None,
+                which='mean'):
+        """
+        Predict response variable of a count model given exogenous variables.
+
+        Notes
+        -----
+        If exposure is specified, then it will be logged by the method.
+        The user does not need to log it first.
+        """
+        if exog is None:
+            exog = self.exog
+        
+        if exposure is None:
+            exposure = getattr(self, 'exposure', 0)
+        elif exposure != 0:
+            exposure = np.log(exposure)
+
+        if offset is None:
+            offset = getattr(self, 'offset', 0)
+
+        fitted = np.dot(exog, params[:exog.shape[1]])
+        linpred = fitted + exposure + offset
+
+        if which == 'mean':
+            return np.exp(linpred)
+        elif which == 'linear':
+            return linpred
+        elif which =='prob':
+            counts = np.atleast_2d(np.arange(0, np.max(self.endog)+1))
+            mu = self.predict(params, exog=exog, exposure=exposure,
+                              offset=offset)[:,None]
+            return genpoisson_p.pmf(counts, mu, params[-2],
+                                    self.parameterization)
```

Comment:
two bugs here in the parameterization, AFAICS
dispersion parameter is `params[-1]`
p parameter is `self.parameterization + 1`